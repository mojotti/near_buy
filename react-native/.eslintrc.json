{
  "extends": "airbnb",
  "rules": {
    "func-names": ["error", "never"],
    "indent": ["error", 2],
    "arrow-parens": "off",
    "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }],
    "react/jsx-indent": ["error", 2],
    "react/jsx-indent-props": ["error", 2],
    "import/no-named-as-default": 0,
    "import/prefer-default-export": 0,
    "no-undef": "off",
    "no-use-before-define": "off"
  },
  "env": {
    "jest": true
  },
  "globals": { "fetch": false }
}
