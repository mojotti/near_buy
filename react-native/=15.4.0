NearBuy@0.1.0 /Users/ristomatti/PycharmProjects/near_buy/react-native
└── UNMET PEER DEPENDENCY react@16.0.0

