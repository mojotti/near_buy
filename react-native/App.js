import React from 'react';
import { Provider } from 'react-redux';
import Application from './components/Application';
import store from './redux';

export default class App extends React.Component {
  render() {
    return (
      <Provider store={store}>
        <Application />
      </Provider>
    );
  }
}
