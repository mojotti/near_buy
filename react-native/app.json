{
  "name": "NearBuy",
  "displayName": "NearBuy",
  "expo": {
    "sdkVersion": "25.0.0"
  }
}