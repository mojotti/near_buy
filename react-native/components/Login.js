'use strict';

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Alert,
    Button,
    Image,
    KeyboardAvoidingView,
    StyleSheet,
    Text,
    TextInput,
    View } from 'react-native';
import { login } from '../redux/actions/auth';
import { generateHeadersForBasicAuth,
    generateHashForRegistering } from '../src/networking';
import { localhost,
    loginText,
    registerText,
    widthWithThirtyPercentPadding} from '../src/static/constants';

export class Login extends Component {
    constructor (props) {
        super(props);
        this.state = {
            page: 'Login',
            username: '',
            password: '',
            email: '',
            textFocused: false
        };
    }

    togglePage () {
        let currentPage = this.getPageTitle(this.state);
        this.setState({ page: currentPage });
    }

    getPageTitle (state) {
        let { page } = state;
        return (page === 'Login') ? 'Sign up' : 'Login';
    }

    getHelperText (state) {
        let { page } = state;
        return (page === 'Login') ? loginText : registerText;
    }

    handleButtonPress(state) {
        const { page, username, password, email } = state;
        if(username === '' || password === '' || (page === 'Sign up' && email === ''))
        {
            Alert.alert('Invalid values', 'Please enter all the values.');
            return;
        }
        if (page === 'Login') {
            this.handleLoginRequest(state)
        } else {
            this.handleRegisteringRequest(state)
        }
    }

    handleRegisteringRequest (state) {
        const { username, password, email } = state;
        let ip_addr = `http://${localhost}:5000/api/v1.0/register`;
        fetch(ip_addr, {
            method: 'POST',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({
                'user_info': generateHashForRegistering(username, password, email)
            })
        })
        .then((response) => response.json())
        .then((responseJson) => {
            if (responseJson.user_creation === 'success') {
                Alert.alert('User creation', 'User created successfully!');
                console.log(responseJson);
                this.setState({ page: 'Login' });
            } else {
                Alert.alert('User creation', 'User exists already!');
            }
        })
        .catch((error) => {
            console.log('error ', error);
            console.error(error);
        });
    }

    handleLoginRequest (state) {
        let ip_addr = `http://${localhost}:5000/api/v1.0/auth`;
        const { username, password } = state;
        fetch(ip_addr, {
            method: 'GET',
            headers: generateHeadersForBasicAuth(username, password)
        })
        .then((response) => response.json())
        .then((responseJson) => {
            console.log(responseJson);
            if (responseJson.username === username) {
                this.props.onLogin(username, responseJson.token);
            } else {
                Alert.alert('Try again, mate!', 'Invalid credentials.');
                this.setState({ password: '' });
            }
        })
        .catch((error) => {
            console.error(error);
        });
    }

    renderEmailInput() {
        if (this.state.page === 'Sign up') {
            return (
                <TextInput
                    placeholder='Email address'
                    autoCapitalize='none'
                    autoCorrect={false}
                    autoFocus={false}
                    keyboardType='email-address'
                    value={this.state.email}
                    onFocus={() => this.setState({textFocused: true})}
                    onChangeText={(text) => this.setState({ email: text })}
                />
            )
        }
    }

    renderLogoAndWelcomeText() {
        if (this.state.textFocused === false) {
            return (
                <View style={[styles.logoContainer]}>
                  <Text style={[styles.welcomeText]}>Welcome to NearBuy</Text>
                  <Image
                      resizeMode="contain"
                      style={styles.logo}
                      source={require('./../logo.png')}
                  />
                </View>
            )
        }
    }

    render () {
        return (
            <View style={[styles.container]}>
              <KeyboardAvoidingView
                  behavior='padding'
                  keyboardVerticalOffset={64}
              >
                  {this.renderLogoAndWelcomeText()}
                <View style={[styles.loginContainer]}>
                    {this.renderEmailInput()}
                  <View style={{margin: 4}}></View>
                  <TextInput
                      placeholder='Username'
                      autoCapitalize='none'
                      autoCorrect={false}
                      autoFocus={false}
                      onFocus={() => this.setState({textFocused: true})}
                      keyboardType='email-address'
                      value={this.state.username}
                      onChangeText={(text) => this.setState({ username: text })} />
                  <View style={{margin: 4}}></View>
                  <TextInput
                      placeholder='Password'
                      autoCapitalize='none'
                      autoCorrect={false}
                      secureTextEntry={true}
                      onFocus={() => this.setState({textFocused: true})}
                      value={this.state.password}
                      onChangeText={(text) => this.setState({ password: text })} />

                  <Text style={{fontSize: 27}}>{this.state.route}</Text>
                  <Button onPress={() => this.handleButtonPress(this.state)} title={this.state.page}/>
                  <View style={{marginTop: 40, flexDirection: 'row', justifyContent: 'center'}}>
                    <Text onPress={() => this.togglePage()} style={{fontSize: 16, color: 'blue'}}>
                        {this.getPageTitle(this.state)}
                    </Text>
                  </View>
                  <Text style={[styles.loginHint]}>{this.getHelperText(this.state)}</Text>
                </View>
              </KeyboardAvoidingView>
            </View>
        );
    }
}


const mapStateToProps = (state, ownProps) => {
    return {
        isLoggedIn: state.auth.isLoggedIn
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        onLogin: (username, token) => { dispatch(login(username, token)); },
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'space-around',
    },
    welcomeText: {
        fontSize: 24,
        textAlign: 'center',
        marginTop: 50
    },
    loginHint: {
        fontSize: 12,
        textAlign: 'center',
        marginTop: 10,
    },
    logo: {
        width: 350,
        height: 150,
    },
    loginContainer: {
        justifyContent: 'center',
        alignSelf: 'center',
        width: widthWithThirtyPercentPadding,
    },
    logoContainer: {
        justifyContent: 'center',
        alignSelf: 'stretch',
        padding: 0
    },
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
