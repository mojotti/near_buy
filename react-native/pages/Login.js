'use strict';

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Alert, ScrollView, Text, TextInput, View, Button, Platform, StyleSheet,
  Image, KeyboardAvoidingView } from 'react-native';
import { login } from '../redux/actions/auth';

const LOCALHOST = (Platform.OS === 'ios') ? 'localhost' : '10.0.2.2';
const base64 = require('base-64');
const loginText = "New user? Press 'Sign up' to register.";
const registerText = "Existing user? Press 'Login'.";


class Login extends Component {
    constructor (props) {
        super(props);
        this.state = {
            page: 'Login',
            username: '',
            password: ''
        };
    }

    userLogin (e) {
        this.props.onLogin(this.state.username, this.state.password);
        e.preventDefault();
    }

    togglePage (e) {
      this.setState({ page: this.altLoginSignup });
      e.preventDefault();
    }

    get altLoginSignup () { return (this.state.page === 'Login')
          ? 'Sign up' : 'Login'; }

    get altHelperText () { return (this.state.page === 'Login')
          ? loginText : registerText; }

    getHeaders() {
      var headers = new Headers();
      var hash = this.getHash();
      headers.append("Authorization", "Basic " + hash);
      return headers;
    }
    
    getHash() {
      var credentials = this.state.username + ':' + this.state.password
      return base64.encode(credentials);
    }

    handleLoginRequest (e) {
      e.preventDefault();
      var headers = this.getHeaders();

      fetch('http://' + LOCALHOST + ':5000/api/v1.0/auth', {
         method: 'GET',
         headers: this.getHeaders()
      })

      .then((response) => response.json())
      .then((responseJson) => {
        console.log(responseJson);
        if (responseJson.login === 'success') {
          this.props.onLogin(this.state.username, this.getHash());
        } else {
          Alert.alert('Try again, mate!', 'Invalid credentials.');
          this.setState({ password: '' });
        }
      })

      .catch((error) => {
         console.error(error);
         return;
      });
    }

    render () {
      return (
          <View style={[styles.container]}>
            <KeyboardAvoidingView
            behavior='padding'
            keyboardVerticalOffset={-70}
            >
            <View style={[styles.logoContainer]}>
              <Text style={[styles.welcomeText]}>Welcome to NearBuy</Text>
              <Image
                resizeMode="contain"
                style={styles.logo}
                source={require('./../logo.png')}
              />
            </View>

            <View style={[styles.loginContainer]}>
            <TextInput
                placeholder='Username'
                autoCapitalize='none'
                autoCorrect={false}
                autoFocus={false}
                keyboardType='email-address'
                value={this.state.username}
                onChangeText={(text) => this.setState({ username: text })} />
            <View style={{margin: 4}}/>
            <TextInput
                placeholder='Password'
                autoCapitalize='none'
                autoCorrect={false}
                secureTextEntry={true}
                value={this.state.password}
                onChangeText={(text) => this.setState({ password: text })} />

              <Text style={{fontSize: 27}}>{this.state.route}</Text>
              <Button onPress={(e) => this.handleLoginRequest(e)} title={this.state.page}/>
              <View style={{marginTop: 55, flexDirection: 'row', justifyContent: 'center'}}>
                <Text onPress={(e) => this.togglePage(e)} style={{fontSize: 14, color: 'blue'}}>
                    {this.altLoginSignup}
                </Text>
              </View>
              <Text style={[styles.loginHint]}>{this.altHelperText}</Text>
          </View>
          </KeyboardAvoidingView>
        </View>
    );
  }
}


const mapStateToProps = (state, ownProps) => {
    return {
        isLoggedIn: state.auth.isLoggedIn
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        onLogin: (username, hash) => { dispatch(login(username, hash)); },
        onSignUp: (username, password) => { dispatch(signup(username, password)); }
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  welcomeText: {
    fontSize: 24,
    textAlign: 'center',
    marginTop: 50
  },
  loginHint: {
    fontSize: 12,
    textAlign: 'center',
    marginTop: 10,
  },
  logo: {
    width: 350,
    height: 150,
  },
  loginContainer: {
    justifyContent: 'center',
    alignSelf: 'stretch',
    padding: 80
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
