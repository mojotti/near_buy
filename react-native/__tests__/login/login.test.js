import React from 'react';
import renderer from 'react-test-renderer';
import 'isomorphic-fetch'; // for headers, fetch, etc.

import { Alert } from 'react-native';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import { Login } from '../../src/components/login/Login';
import {
  loginText,
  registerText,
  SUCCESSFUL_REGISTRATION_ALERT,
  USER_EXISTS_ALERT,
  MISSING_USER_DETAILS_ALERT,
  NETWORK_ERROR_ALERT,
  INVALID_CREDS_ALERT,
  NETWORK_ERROR,
} from '../../src/static/constants';
import {
  successfulRegisteringResponse,
  successfulLoginResponse,
  unsuccessfulLoginResponse,
  userExistsRegisteringResponse,
} from '../__mocks__/loginAndRegisteringSamples';

const onLoginSpy = sinon.spy();
const mockedProps = {
  onLogin: onLoginSpy,
};

describe('Login', () => {
  let loginComponent;

  beforeEach(() => {
    Alert.alert = jest.fn();
    loginComponent = shallow(<Login {...mockedProps} />);
  });

  test('given username is missing, when sign-up-button is pressed, then alert is shown', () => {
    loginComponent.setState({
      page: 'Sign up',
      username: '',
      password: 'bar',
      email: 'foo@bar.com',
    });

    const render = loginComponent.find('LoginAndRegisterButton').dive();
    const loginButton = render.find('[id="loginButton"]');
    loginButton.simulate('Press');

    expect(Alert.alert.mock.calls.length).toBe(1);
    expect(Alert.alert.mock.calls).toEqual([MISSING_USER_DETAILS_ALERT]);
  });

  test('given email is missing, when sign-up-button is pressed, then alert is shown', () => {
    loginComponent.setState({
      page: 'Sign up',
      username: 'foo',
      password: 'bar',
      email: '',
    });

    const render = loginComponent.find('LoginAndRegisterButton').dive();
    const loginButton = render.find('[id="loginButton"]');
    loginButton.simulate('Press');

    expect(Alert.alert.mock.calls.length).toBe(1);
    expect(Alert.alert.mock.calls).toEqual([MISSING_USER_DETAILS_ALERT]);
  });

  test('given pw is missing, when sign-up-button is pressed, then alert is shown', () => {
    loginComponent.setState({
      page: 'Sign up',
      username: 'foo',
      password: '',
      email: 'foo@bar.com',
    });

    const render = loginComponent.find('LoginAndRegisterButton').dive();
    const loginButton = render.find('[id="loginButton"]');
    loginButton.simulate('Press');

    expect(Alert.alert.mock.calls.length).toBe(1);
    expect(Alert.alert.mock.calls).toEqual([MISSING_USER_DETAILS_ALERT]);
  });

  test('given username is missing, when login-button is pressed, then alert is shown', () => {
    loginComponent.setState({
      page: 'Login',
      username: '',
      password: 'bar',
      email: 'foo@bar.com',
    });

    const render = loginComponent.find('LoginAndRegisterButton').dive();
    const loginButton = render.find('[id="loginButton"]');
    loginButton.simulate('Press');

    expect(Alert.alert.mock.calls.length).toBe(1);
    expect(Alert.alert.mock.calls).toEqual([MISSING_USER_DETAILS_ALERT]);
  });

  test('alert is raised when network error occurs', () => {
    const error = {
      message: 'Network request failed',
    };
    loginComponent.instance().handleError(error);

    expect(Alert.alert.mock.calls.length).toBe(1);
    expect(Alert.alert.mock.calls).toEqual([NETWORK_ERROR_ALERT]);
  });

  test('given pw is missing, when login-button is pressed, then alert is shown', () => {
    fetch.mockResponseSuccess(successfulLoginResponse);

    loginComponent.setState({
      page: 'Login',
      username: 'foo',
      password: '',
    });
    const render = loginComponent.find('LoginAndRegisterButton').dive();

    const loginButton = render.find('[id="loginButton"]');
    loginButton.simulate('Press');

    expect(Alert.alert.mock.calls.length).toBe(1);
    expect(Alert.alert.mock.calls).toEqual([MISSING_USER_DETAILS_ALERT]);
  });

  test('given all details are missing, when login-button is pressed, then alert is shown', () => {
    loginComponent.setState({
      page: 'Login',
      username: '',
      password: '',
      email: '',
    });
    const render = loginComponent.find('LoginAndRegisterButton').dive();

    const loginButton = render.find('[id="loginButton"]');
    loginButton.simulate('Press');

    expect(Alert.alert.mock.calls.length).toBe(1);
    expect(Alert.alert.mock.calls).toEqual([MISSING_USER_DETAILS_ALERT]);
  });

  test('given state is login, when login-button is pressed, then alert is not shown', () => {
    // to avoid errors in fetch
    fetch.mockResponseSuccess(unsuccessfulLoginResponse);

    loginComponent.setState({
      page: 'Login',
      username: 'foo',
      password: 'bar',
      email: 'foo@bar.com',
    });

    const render = loginComponent.find('LoginAndRegisterButton').dive();
    const loginButton = render.find('[id="loginButton"]');
    loginButton.simulate('Press');

    // alert would be risen even before fetch
    expect(Alert.alert.mock.calls.length).toBe(0);
  });

  test('when sign-up is successful, alert is shown', () => {
    fetch.mockResponseSuccess(successfulRegisteringResponse);

    loginComponent.setState({
      page: 'Sign up',
      username: 'foo',
      password: 'bar',
      email: 'foo@bar.com',
    });

    loginComponent.update();
    const render = loginComponent.find('LoginAndRegisterButton').dive();
    const loginButton = render.find('[id="loginButton"]');

    // just to see if errors in fetch
    loginButton.simulate('Press');

    // testing the actual fetch logic
    loginComponent
      .instance()
      .handleRegisteringResponse(successfulRegisteringResponse);

    expect(loginComponent.state('page')).toEqual('Login');
    expect(Alert.alert.mock.calls.length).toBe(1);
    expect(Alert.alert.mock.calls).toEqual([SUCCESSFUL_REGISTRATION_ALERT]);
  });

  test('when existing user is tried to be registered, alert is shown and registration fails', () => {
    fetch.mockResponseSuccess(userExistsRegisteringResponse);

    loginComponent.setState({
      page: 'Sign up',
      username: 'foo',
      password: 'bar',
      email: 'foo@bar.com',
    });

    loginComponent.update();
    const render = loginComponent.find('LoginAndRegisterButton').dive();
    const loginButton = render.find('[id="loginButton"]');

    // just to see if errors in fetch
    loginButton.simulate('Press');

    // testing the actual fetch logic
    loginComponent
      .instance()
      .handleRegisteringResponse(userExistsRegisteringResponse);

    expect(loginComponent.state('page')).toEqual('Sign up');
    expect(Alert.alert.mock.calls.length).toBe(1);
    expect(Alert.alert.mock.calls).toEqual([USER_EXISTS_ALERT]);
  });

  test('when error is risen in registering, it is handled', () => {
    fetch.mockResponseFailure(NETWORK_ERROR);

    loginComponent.setState({
      page: 'Sign up',
      username: 'foo',
      password: 'bar',
      email: 'foo@bar.com',
    });

    loginComponent.update();
    const render = loginComponent.find('LoginAndRegisterButton').dive();
    const loginButton = render.find('[id="loginButton"]');

    // just to see if errors in fetch
    loginButton.simulate('Press');
  });

  test('when error is risen in login, it is handled', () => {
    fetch.mockResponseFailure([NETWORK_ERROR]);

    loginComponent.setState({
      page: 'Login',
      username: 'foo',
      password: 'bar',
    });

    loginComponent.update();
    const render = loginComponent.find('LoginAndRegisterButton').dive();
    const loginButton = render.find('[id="loginButton"]');

    // just to see if errors in fetch
    loginButton.simulate('Press');
  });

  test('when logging in with invalid creds, alert is shown and login fails', () => {
    fetch.mockResponseSuccess(unsuccessfulLoginResponse);

    loginComponent.setState({
      page: 'Sign up',
      username: 'foo',
      password: 'bar',
      email: 'foobar',
    });

    const render = loginComponent.find('LoginAndRegisterButton').dive();
    const loginButton = render.find('[id="loginButton"]');

    // just to see if errors in fetch
    loginButton.simulate('Press');

    // testing the actual fetch logic
    loginComponent.instance().handleLoginResponse(unsuccessfulLoginResponse);

    expect(loginComponent.state('password')).toEqual('');
    expect(Alert.alert.mock.calls.length).toBe(1);
    expect(Alert.alert.mock.calls).toEqual([INVALID_CREDS_ALERT]);
  });

  test('when logging in with correct creds, user is logged in', () => {
    fetch.mockResponseSuccess(successfulLoginResponse);

    loginComponent.setState({
      page: 'Sign up',
      username: 'foo',
      password: 'bar',
      email: 'foobar',
    });

    const render = loginComponent.find('LoginAndRegisterButton').dive();
    const loginButton = render.find('[id="loginButton"]');

    // just to see if errors in fetch
    loginButton.simulate('Press');

    // testing the actual fetch logic
    loginComponent.instance().handleLoginResponse(successfulLoginResponse);

    expect(loginComponent.state('password')).toEqual('bar');
    expect(onLoginSpy.called).toBeTruthy();
    expect(onLoginSpy.args[0][0]).toEqual(successfulLoginResponse.username);
    expect(onLoginSpy.args[0][1]).toEqual(successfulLoginResponse.token);
  });

  test('Given current page is login, when page title is requested, then title is Sign up', () => {
    loginComponent.setState({
      page: 'Login',
      username: '',
      password: '',
      email: '',
    });

    const render = loginComponent.find('AltPage').dive();
    const button = render.find('[id="altPageTitle"]');
    expect(button.props().children).toBe('Sign up');
  });

  test('Given current page is sign up, when page title is requested, then title is Login', () => {
    loginComponent.setState({
      page: 'Sign up',
      username: '',
      password: '',
      email: '',
    });

    const render = loginComponent.find('AltPage').dive();
    const button = render.find('[id="altPageTitle"]');
    expect(button.props().children).toBe('Login');
  });

  test('Given current page is login, when helper text is requested, then text is login', () => {
    loginComponent.setState({
      page: 'Login',
      username: '',
      password: '',
      email: '',
    });

    const render = loginComponent.find('AltPage').dive();
    const button = render.find('[id="helperText"]');
    expect(button.props().children).toBe(loginText);
  });

  test('Given current page is sign up, when helper text is requested, then text is sign up', () => {
    loginComponent.setState({
      page: 'Sign up',
      username: '',
      password: '',
      email: '',
    });

    const render = loginComponent.find('AltPage').dive();
    const button = render.find('[id="helperText"]');
    expect(button.props().children).toBe(registerText);
  });

  test('renders sign up details when state is sign up', () => {
    const loginComp = renderer.create(<Login {...mockedProps} />);
    expect(loginComp.getInstance().state.page).toEqual('Login');

    loginComp.getInstance().togglePage();
    expect(loginComp.getInstance().state.page).toEqual('Sign up');

    const tree = loginComp.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Logo is hidden when keyboard is visible', () => {
    const loginComp = renderer.create(<Login {...mockedProps} />);
    loginComp.getInstance().keyboardDidShow();

    expect(loginComp.getInstance().state.isKeyboardVisible).toBeTruthy();

    const tree = loginComp.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('isKeyboardVisible is set to false when keyboard is hidden', () => {
    expect(loginComponent.state('isKeyboardVisible')).toBeFalsy();

    loginComponent.setState({ isKeyboardVisible: true });
    loginComponent.instance().keyboardDidHide();

    expect(loginComponent.state('isKeyboardVisible')).toBeFalsy();
  });

  test('credential change is handled correctly', () => {
    expect(loginComponent.state('username')).toEqual('');
    expect(loginComponent.state('password')).toEqual('');

    loginComponent
      .instance()
      .handleCredentialChange({ username: 'foo', password: null });

    expect(loginComponent.state('username')).toEqual('foo');
    expect(loginComponent.state('password')).toEqual('');

    loginComponent
      .instance()
      .handleCredentialChange({ username: null, password: 'bar' });

    expect(loginComponent.state('username')).toEqual('foo');
    expect(loginComponent.state('password')).toEqual('bar');
  });

  test('email is changed', () => {
    expect(loginComponent.state('email')).toEqual('');

    loginComponent.instance().handleEmailChange('foo@bar.com');

    expect(loginComponent.state('email')).toEqual('foo@bar.com');
  });
});
