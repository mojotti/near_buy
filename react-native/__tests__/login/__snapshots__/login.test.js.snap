// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Logo is hidden when keyboard is visible 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-around",
      },
    ]
  }
>
  <View
    keyboardVerticalOffset={64}
    onLayout={[Function]}
    style={
      Array [
        undefined,
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "justifyContent": "center",
          "width": 525,
        }
      }
    >
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          autoFocus={false}
          id="usernameTextInput"
          onChangeText={[Function]}
          placeholder="Username"
          style={
            Object {
              "marginBottom": 12,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          id="passwordTextInput"
          onChangeText={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          style={
            Object {
              "marginBottom": 16,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4d4dff",
            "borderRadius": 30,
            "elevation": 1,
            "paddingBottom": 8,
            "paddingTop": 8,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontFamily": null,
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Login
        </Text>
      </View>
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 40,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            id="altPageTitle"
            onPress={[Function]}
            style={
              Object {
                "color": "blue",
                "fontSize": 16,
              }
            }
          >
            Sign up
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          id="helperText"
          style={
            Object {
              "fontFamily": null,
              "fontSize": 12,
              "marginTop": 10,
              "textAlign": "center",
            }
          }
        >
          New user? Press 'Sign up' to register.
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Login renders sign up details when state is sign up 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-around",
      },
    ]
  }
>
  <View
    keyboardVerticalOffset={64}
    onLayout={[Function]}
    style={
      Array [
        undefined,
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
          "padding": 0,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontFamily": null,
            "fontSize": 24,
            "marginTop": 50,
            "textAlign": "center",
          }
        }
      >
        Hi! Welcome to
      </Text>
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/static/images/logo.png",
          }
        }
        style={
          Object {
            "height": 213.44,
            "marginBottom": 213.44,
            "width": 600,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "justifyContent": "center",
          "width": 525,
        }
      }
    >
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          autoFocus={false}
          id="emailTextInput"
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="Email address"
          style={
            Object {
              "marginBottom": 12,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          autoFocus={false}
          id="usernameTextInput"
          onChangeText={[Function]}
          placeholder="Username"
          style={
            Object {
              "marginBottom": 12,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          id="passwordTextInput"
          onChangeText={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          style={
            Object {
              "marginBottom": 16,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4d4dff",
            "borderRadius": 30,
            "elevation": 1,
            "paddingBottom": 8,
            "paddingTop": 8,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontFamily": null,
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Sign up
        </Text>
      </View>
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 40,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            id="altPageTitle"
            onPress={[Function]}
            style={
              Object {
                "color": "blue",
                "fontSize": 16,
              }
            }
          >
            Login
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          id="helperText"
          style={
            Object {
              "fontFamily": null,
              "fontSize": 12,
              "marginTop": 10,
              "textAlign": "center",
            }
          }
        >
          Existing user? Press 'Login'.
        </Text>
      </View>
    </View>
  </View>
</View>
`;
